src/main/scala/com/atomist/tree/pathexpression/ExecutionResult.scala

```diff
@@ -0,0 +1,50 @@
+package com.atomist.tree.pathexpression
+
+import com.atomist.graph.{AddressableGraphNode, GraphNode}
+
+/**
+  * Result type and utility methods of a node navigation.
+  */
+object ExecutionResult {
```

Comment:
Do we execute path expressions or evaluate them? I'm wondering if Execution is a little misleading.